modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java

```diff
@@ -626,6 +666,9 @@ private synchronized void _generate(
 		_createFDSFieldObjectDefinition(
 			fdsViewObjectDefinition, locale, userId);
 		_createFDSSortObjectDefinition(fdsViewObjectDefinition, locale, userId);
+
+		_objectDefinitionLocalService.publishSystemObjectDefinition(
```

Comment:
It seems that we only need to publish the root object and it will recursively will drill down the hierarchy and publish all descendants.
However, publication is not working as expected. There are some differences between [publishCustomObjectDefinition](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java#L820) and [publishSystemObjectDefinition](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java#L856)